lib/engine/route.rb

```diff
@@ -123,16 +123,22 @@ def touch_node(node)
             connection_data << segment(connection, left: tail[:right])
           end
         end
+        connection_data.pop if @train.local? && connection_data.size == 2
       elsif @last_node == node
         @last_node = nil
+        connection_data.clear if @train.local? && !connection_data.empty?
```

Comment:
do you need to check connection_data here? why not just always clear it?

Reply:
No its true we can always clear it here.